ext/coverage/coverage.c

```diff
@@ -75,6 +75,35 @@ rb_coverage_result(VALUE klass)
     return ncoverages;
 }
 
+static int
+coverage_merge_result_i(st_data_t key, st_data_t val, st_data_t h)
+{
+    VALUE path      = (VALUE)key;
+    VALUE coverage  = (VALUE)val;
+    VALUE coverages = (VALUE)h;
+    VALUE ary;
+
+    ary = rb_hash_lookup2(coverages, path, Qnil);
+
+    if (RTEST(ary)) {
+      rb_ary_replace(ary, coverage);
+    }
+
+    return ST_CONTINUE;
+}
+
+static VALUE
+rb_coverage_set_result(VALUE klass, VALUE ncoverages)
+{
+    VALUE coverages = rb_get_coverages();
+    if (!RTEST(coverages)) {
+	rb_raise(rb_eRuntimeError, "coverage measurement is not enabled");
+    }
+    st_foreach(RHASH_TBL(ncoverages), coverage_merge_result_i, coverages);
```

Comment:
Should check if the argument is a `Hash` before `RHASH_TBL`.
`Coverage.start; Coverage.result = nil` segfaults.